maintenance/kubernetes/aks-1.22.md

```diff
@@ -0,0 +1,186 @@
+---
+tags: maintenance
+---
+<!-- markdownlint-disable MD013 MD036-->
+
+# Jenkins AKS Kubernetes Upgrade to 1.22
+
+https://docs.microsoft.com/en-us/azure/aks/upgrade-cluster?tabs=azure-cli
+
+## 1.21.x  to 1.22.x
+
+Related to `prodpublicks8` and `temp-privatek8s` clusters
+
+- [x] Issue: <https://github.com/jenkins-infra/helpdesk/issues/2930>
+
+- [X] Announces:
+  - [X] `status.jenkins.io`: <https://github.com/jenkins-infra/status/pull/99>
+  - [ ] IRC
+  - [x] ML (jenkins-infra@googlegroups.com, jenkinsci-dev@googlegroups.com) [template bellow]
+
+- [X] Changelog Scrapping:
+  - [x] [Urgent Upgrade Notes for 1.22](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md#urgent-upgrade-notes), nothing to report
+
+  - [x] Azure:
+    - <https://github.com/Azure/AKS/blob/master/CHANGELOG.md>: nothing to report
+
+  - [x] Docker Helmfile Upgrade (kubectl):
+    - [x] <https://github.com/jenkins-infra/docker-helmfile/pull/>
+  
+  - [x] Checking for deprecated APIs on Kube resources/helmfiles
+  
+  ```shell
+  $ pluto detect-files -d ./charts
+  There were no resources found with known deprecated apiVersions.
+  ```
+
+  - Upgrade Procedure:
+    - [x] Ensure we have access to all clusters in kube-config
+    - [x] Docker Helmfile Upgrade (kubectl)
+    - [x] Merging last PRs on charts or putting it as on-hold
+    - [x] Pagerduty: Notify the on call person
+    - [ ] Stopping the k8s management job
```

Comment:
```suggestion
    - [x] Stopping the k8s management job
```
It's the new step we're adding, let's say it's been done ^^

Reply:
it's a lie 🤣